kornia/augmentation/random_generator/_2d/affine.py

```diff
@@ -109,23 +109,23 @@ def make_samplers(self, device: torch.device, dtype: torch.dtype) -> None:
         shear_y_sampler: Optional[Uniform] = None
 
         if _translate is not None:
-            translate_x_sampler = Uniform(-_translate[0], _translate[0], validate_args=False)
-            translate_y_sampler = Uniform(-_translate[1], _translate[1], validate_args=False)
+            translate_x_sampler = Uniform(-_translate[0].clone(), _translate[0].clone(), validate_args=False)
```

Comment:
can't we wrap a `UniformDistribution` class that handles the clone method instead of having to manually do that everyone it's implemented ? /cc @shijianjian @johnnv1

Reply:
makes sense to me, this would make it easier for future contributors not to make the same mistake

Reply:
Hi! Thanks for the feedback. It makes sense to me. Do you have a suggestion for where this distribution should leave in the code base? Also there are a number of cases where the tensors are directly stored as attributes. I would imagine I could solve this in a generic fashion using `__setattr__`. Do you have any opinion on this? See for example [this](https://github.com/kornia/kornia/pull/2499/files#diff-ddcd748535c88a50fb49c1b4ee19f0ed10096a065a23267b83be11c2cb57e890R61)

Reply:
In the end, given the the direct assignment of tensors only occurs once I decided against making things more complex with `__setattr__`.  I move the UniformDistribution to base given there already was another class derived from torch `Distribution` there.